Changes to Success Modal after first setup

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

Closes #555

- Clone the Guide of `@wordpress/components` to extend features (ref: https://github.com/woocommerce/google-listings-and-ads/issues/555#issuecomment-843125024)
    - Add a render function for the cloned Guide to customize the finish block
    - Add a prop for the cloned Guide to customize the button text
- Add a "Create paid campaign" button to the success modal in the product feed page
- Update contents and style to meet the new changes of the success modal

### Screenshots:

#### First page

![image](https://user-images.githubusercontent.com/17420811/118781681-e7135400-b8bf-11eb-8c0d-5ffaf220eb03.png)

#### Second page

![image](https://user-images.githubusercontent.com/17420811/118781699-ec709e80-b8bf-11eb-8d44-8a4c33dca768.png)

#### Demo GIF

![Kapture 2021-05-19 at 16 13 56](https://user-images.githubusercontent.com/17420811/118779786-f98c8e00-b8bd-11eb-9763-d6e819abe640.gif)

### Detailed test instructions:

1. Open the DevTool console and execute `localStorage.setItem( 'debug', 'wc-admin:*' );` to enable tracking debugging mode
2. Go to product feed page with opened success modal: `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fproduct-feed&guide=submission-success`
3. The contents of success modal should fit with the visual design
4. Click on the "Manage and edit your product feed in WooCommerce" link should open the product feed page in a new tab
5. Click on the "Create paid campaign" button should go to the ads campaign setup page in the current tab
6. Click on the "Maybe later" button should close the success modal
7. The related events should be logged on the DevTool console

### Changelog Note:

> Update contents and style, and add a "Create paid campaign" button to the success modal in the product feed page

eason9487 committed:
Clone the Guide of @wordpress/components to extend features

eason9487 committed:
Remove deprecated check from the cloned Guide component

eason9487 committed:
Add JSDoc to the cloned Guide component

eason9487 committed:
Add a render function for the cloned Guide to customize the finish block

eason9487 committed:
Add "Create paid campaign" button to the success modal in the product feed page

eason9487 committed:
Add a prop for the cloned Guide to customize the  button text

eason9487 committed:
Update contents and style to meet the new changes of the success modal

j111q reviewed (approved):
Looking great! Thanks @eason9487! 🚀

ecgan reviewed (commented):
Tested in my local machine, seems to work okay in terms of functionality. Left a few comments about track events. Also, I think there are some unused code which can be cleaned up.

eason9487 committed:
Use `children` instead of another `buttonText` props to customize the button text

Address https://github.com/woocommerce/google-listings-and-ads/pull/632/files#r638534167

eason9487 committed:
Change the `action` of tracking event props to meet the actual meaning of "Maybe later" button

Address https://github.com/woocommerce/google-listings-and-ads/pull/632/files#r638558382

ecgan reviewed (approved):
LGTM 👍
